harness/determined/common/storage/util.py

```diff
@@ -0,0 +1,46 @@
+import urllib
+
+from determined.common import storage
+
+
+def from_string(shortcut) -> storage.base.StorageManager:
+    p: urllib.parse.ParseResult = urllib.parse.urlparse(shortcut)
+    if p.scheme == "ms":
+        container = p.netloc
+        connection_string = p.fragment
+        kwargs = urllib.parse.parse_qs(p.query)
+        account_url = kwargs.get("account_url", [None])[0]
+        credential = kwargs.get("credential", [None])[0]
+        temp_dir = kwargs.get("temp_dir", [None])[0]
+        return storage.AzureStorageManager(
+            container=container,
+            connection_string=connection_string,
+            account_url=account_url,
+            credential=credential,
+            temp_dir=temp_dir,
+        )
+    elif p.scheme == "gs":
+        bucket = p.netloc
+        prefix = p.path.lstrip("/")
+        kwargs = urllib.parse.parse_qs(p.query)
+        temp_dir = kwargs.get("temp_dir", [None])[0]
+        print(f"{kwargs=}")
+        return storage.gcs.GCSStorageManager(bucket=bucket, prefix=prefix, temp_dir=temp_dir)
+    elif p.scheme == "s3":
+        bucket = p.netloc
+        prefix = p.path.lstrip("/")
+        kwargs = urllib.parse.parse_qs(p.query)
+        access_key = kwargs.get("access_key", [None])[0]
+        secret_key = kwargs.get("secret_key", [None])[0]
```

Comment:
are these part of the actual s3 url specification, or custom extensions to it?

Reply:
The latter, as far as I know. [Their specification](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html) only mentions buckets and key names.

Reply:
Since these shortcut strings are for instantiating our own classes, I suppose it makes sense that we'd extend or (in the case of Azure) reinvent the URIs.

Reply:
I would rather we stick to the default URIs until we receive customer feedback that they actually want these custom extensions.

My reasoning is that I want people to feel like it's obvious how to use the system, and "determined's s3 uris are just standard s3 uris" is a cheap and easy way to do that.

I think the extensions we offer are probably not widely used.  And if it turns out they are, then we can add extensions later, no harm done.  But it's very hard to go the other way; features that have been cast into the wild are very difficult to reel back in.

Reply:
If there's no existing scheme for Azure, let's just not support Azure shortcut strings yet.  I think we might only have like one Azure user that I've heard of.